osu.Game/Screens/OnlinePlay/Lounge/Components/RoomsContainer.cs

```diff
@@ -141,9 +141,18 @@ private void addRooms(IEnumerable<Room> rooms)
 
         private void removeRooms(IEnumerable<Room> rooms)
         {
-            foreach (var r in rooms)
+            foreach (var room in rooms)
             {
-                roomFlow.RemoveAll(d => d.Room == r, true);
+                var drawableRoom = roomFlow.SingleOrDefault(d => d.Room == room);
+                if (drawableRoom == null)
+                    continue;
+
+                // expire to trigger async disposal. the room still has to exist somewhere so we move it to internal content of RoomsContainer until next frame.
```

Comment:
What on earth is this for? This is some unprecendented hackery. Why do the items need to be moved anywhere like that? Can you explain?

I am not having this solution. I would rather consider something like expiring the entire flow or something rather than doing something this cursed.

Reply:
Well, in order for async disposal to kick in, the drawable needs to be expired first, then the next time `UpdateChildrenLife` is called (i.e. next frame), it's marked dead and scheduled for async disposal.

I could change this to `Clear` when all rooms are removed, since that's another method of triggering async disposal, and seemingly every time `removeRange` is called in real life scenarios, it's always a clear operation [invoked by `ListingPollingComponent` on filter change](https://github.com/ppy/osu/blob/7bc8908ca9c026fed1d831eb6e58df7624a8d614/osu.Game/Screens/OnlinePlay/Components/ListingPollingComponent.cs#L30).

Reply:
If `Clear()` can work then please let's do that. Separate method though please, no weird "check if all items need to be removed and if so clear otherwise removerange" conditionals if it can be avoided.

Reply:
It has to be done at least in the callback of `CollectionsChanged`, the bindable list flow unfortunately doesn't trigger a `Reset` action when clearing all items.